### PR TITLE
Add ansible and ansible vault, update deploy script

### PR DESCRIPTION
**Pull request by billoneil**
Not using the vault secrets yet but it seems to be working.

### CODE REVIEW

File: ansible/roles/apps/stubbornjava/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+  - name: Creating App Directory
+    file:
+      path: apps/stubbornjava
```
Comment (billoneil) [inline ansible/roles/apps/stubbornjava/tasks/main.yml]: I wanted to do `/apps/{appName}` but it needs better permissions. We can set up users per service and reorganize later.

File: ansible/roles/apps/stubbornjava/vars/main.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+  supervisor:
+    app_name: stubbornjava
+    app_directory: /home/ec2-user/apps/stubbornjava
+    app_command: java8 -Denv=prod -Xmx640m -cp 'stubbornjava-all.jar' com.stubbornjava.webapp.StubbornJavaWebApp
```
Comment (billoneil) [inline ansible/roles/apps/stubbornjava/vars/main.yml]: The java8 role doesn't default to java8 for `java` but I kind of like this explicit way more so I didn't bother fixing the default.

File: ansible/roles/common/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+- name: EPEL
+  yum: name=epel-release state=present
+
+- name: New York Time Zone
+  file: src=/usr/share/zoneinfo/America/New_York dest=/etc/localtime owner=root group=root state=link force=true
```
Comment (billoneil) [inline ansible/roles/common/tasks/main.yml]: Is this even right? good / bad idea?

File: ansible/roles/supervisord/files/supervisord.conf [hunk 4]
```diff
@@ -0,0 +1,139 @@
+; Sample supervisor config file.
```
Comment (billoneil) [inline ansible/roles/supervisord/files/supervisord.conf]: Anything obviously wrong here? I want supervisorctl to only be accessible from the server. If we enable the web mode we will need to use vault to hide the user / password.

File: ansible/roles/supervisord/handlers/main.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+---
+- name: restart supervisord
+  service: name=supervisord state=restarted
```
Comment (billoneil) [inline ansible/roles/supervisord/handlers/main.yml]: This should eventually add a handler to just restart the app not supervisor itself. Theres also a way to reload configs but this is ok for now.

File: ansible/stubbornjava.yml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# ansible-playbook --vault-password-file .vault_pw.txt -i inventories/production stubbornjava.yml
+---
+- hosts: stubbornjava
+  become: true
+  roles:
+    - { role: common, become: true }
+    - { role: java, become: true }
+    - { role: apps/stubbornjava, become: true }
```
Comment (billoneil) [inline ansible/stubbornjava.yml]: I tried to only turn sudo on for the roles that needed it but they all needed it for at least one step 🤷‍♂️

File: stubbornjava-webapp/deploy.sh [hunk 7]
```diff
@@ -3,5 +3,5 @@
 # Ideally it would keep versioned jars and symlink the current version for easy rollbacks
 # Another option is to use AWS code build / deploy eventually
 cd .. && gradle clean shadowJar &&
-ansible -i ../stubbornjava/ansible/hosts stubbornjava -m copy -a "src=stubbornjava-webapp/build/libs/stubbornjava-all.jar dest=~/" &&
-ansible -i ../stubbornjava/ansible/hosts stubbornjava -m command -a "supervisorctl restart all"
+ansible --vault-password-file ansible/.vault_pw.txt -i ansible/inventories/production stubbornjava -m copy -a "src=stubbornjava-webapp/build/libs/stubbornjava-all.jar dest=~/apps/stubbornjava" &&
+ansible --vault-password-file ansible/.vault_pw.txt -i ansible/inventories/production stubbornjava -m command -a "supervisorctl restart stubbornjava"
```
Comment (billoneil) [inline stubbornjava-webapp/deploy.sh]: I think there should be a way to make a playbook role that does this. Good enough for now though.

File: ansible/inventories/production/hosts [hunk 8]
```diff
@@ -0,0 +1,2 @@
+[stubbornjava]
+54.84.142.75
```
Comment (rodriguezsergio) [inline ansible/inventories/production/hosts]: You should use the ec2 inventory script in the future.
http://docs.ansible.com/ansible/latest/intro_dynamic_inventory.html#example-aws-ec2-external-inventory-script

Comment (billoneil) [reply ansible/inventories/production/hosts]: Once we get terraform working I'll do that.

File: ansible/roles/common/tasks/main.yml (hunk 3 above)
Comment (rodriguezsergio) [reply ansible/roles/common/tasks/main.yml]: Yeah, that's fine.

File: ansible/roles/java/vars/main.yml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+---
+  java_packages:
+    - java-1.8.0-openjdk
```
Comment (rodriguezsergio) [inline ansible/roles/java/vars/main.yml]: You can override variables in a playbook without creating an additional role.

Comment (billoneil) [reply ansible/roles/java/vars/main.yml]: I refactored that out in the next PR. I thought I was going to need more than just this

File: ansible/roles/supervisord/files/supervisord.conf (hunk 4 above)
Comment (rodriguezsergio) [reply ansible/roles/supervisord/files/supervisord.conf]: The docs say if the `inet_http_server` section is omitted, then supervisor will not start the HTTP server so yeah you're fine.

File: ansible/stubbornjava.yml [hunk 10]
```diff
@@ -0,0 +1,13 @@
+# ansible-playbook --vault-password-file .vault_pw.txt -i inventories/production stubbornjava.yml
+---
+- hosts: stubbornjava
+  become: true
+  roles:
+    - { role: common, become: true }
+    - { role: java, become: true }
+    - {
+        role: apps/app_base,
```
[the remark this answers is not in the document]
Comment (rodriguezsergio) [reply ansible/stubbornjava.yml]: `become: true` seems unnecessary so many times.

[the remark this answers is not in the document]
Comment (billoneil) [reply ansible/stubbornjava.yml]: I wanted to try and only use sudo where needed but I ended up needing it in each role. I think there is a way to just sudo an individual task so I might look into that.

File: stubbornjava-webapp/deploy.sh [hunk 11]
```diff
@@ -3,5 +3,5 @@
 # Ideally it would keep versioned jars and symlink the current version for easy rollbacks
 # Another option is to use AWS code build / deploy eventually
 cd .. && gradle clean shadowJar &&
-ansible -i ../stubbornjava/ansible/hosts stubbornjava -m copy -a "src=stubbornjava-webapp/build/libs/stubbornjava-all.jar dest=~/" &&
-ansible -i ../stubbornjava/ansible/hosts stubbornjava -m command -a "supervisorctl restart all"
+ansible --vault-password-file ansible/.vault_pw.txt -b -i ansible/inventories/production stubbornjava -m copy -a "src=stubbornjava-webapp/build/libs/stubbornjava-all.jar dest=/apps/stubbornjava owner=stubbornjava group=stubbornjava" &&
```
[the remark this answers is not in the document]
Comment (rodriguezsergio) [reply stubbornjava-webapp/deploy.sh]: 👏

**Pull request merged**